translate/src/hooks/usePluralExamples.ts

```diff
@@ -0,0 +1,52 @@
+import type { LocaleType } from '~/context/locale';
+
+const cache = new WeakMap<LocaleType, Record<number, number>>();
+
+/**
+ * A map of examples for the current locale's cldrPlurals.
+ *
+ * For example, for Slovenian (sl) the function returns the following:
+ *     {
+ *          1: 1,
+ *          2: 2,
+ *          3: 3,
+ *          5: 0,
+ *     }
+ *
+ * @returns A map of locale's cldrPlurals and their plural examples.
+ */
+export function usePluralExamples(locale: LocaleType): Record<number, number> {
```

Comment:
This isn't a hook anymore, right?

I think we should move this into core/plurals

Reply:
Actually, on further thought, it should probably be made into a hook, dropping the custom `cache` and relying on `useMemo` instead.